lib/buffer.js

```diff
@@ -326,13 +326,34 @@ Object.defineProperty(Buffer.prototype, 'offset', {
 function slowToString(encoding, start, end) {
   var loweredCase = false;
 
-  start = start >>> 0;
-  end = end === undefined || end === Infinity ? this.length : end >>> 0;
+  if (start >= this.length || end <= 0 || Number.isNaN(end))
```

Comment:
You're special casing for `NaN`. i.e. `Number.isNaN(false) === false`. Are you thinking `isNaN()`?